meson.build

```diff
@@ -0,0 +1,277 @@
+fontconfig = fontconfig_dep.found()
+if fontconfig
+    deps += fontconfig_dep
+    conf.set('CONFIG_FONTCONFIG', 1)
+endif
+
+if get_option('test')
+    test_deps += dependency('libpng', version: '>= 1.2.0')
+    conf.set('CONFIG_LIBPNG', 1)
+endif
+
+# DirectWrite
+directwrite = false
+if host_system == 'windows' and not get_option('directwrite').disabled()
+    if cc.has_header('windows.h')
+        directwrite = true
+        conf.set('CONFIG_DIRECTWRITE', 1)
+        code = '''#include <windows.h>
+            #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+            #error Win32 desktop APIs are available
+            #endif'''
+        if cc.compiles(code)
+            # WinRT/UWP/app build: GDI and LoadLibrary are unavailable,
+            # but DirectWrite is always present
+            deps += cc.find_library('dwrite', required: true)
+        else
+            # Win32/desktop build: GDI is always present;
+            # DirectWrite is optional but can be loaded via LoadLibrary
+            deps += cc.find_library('gdi32', required: true)
+        endif
+    elif get_option('directwrite').enabled()
+        error('DirectWrite was enabled explicitly, but required header is missing.')
+    endif
+endif
+
+# CoreText
+coretext = false
+if host_system == 'darwin' and not get_option('coretext').disabled()
+    coretext_dep = dependency('appleframeworks', modules: ['CoreText', 'CoreFoundation'], required: false)
+    appservices_dep = dependency('appleframeworks', modules: ['ApplicationServices', 'CoreFoundation'], required: false)
+    if cc.has_type('CTRunRef', prefix: '#include <CoreText/CoreText.h>', dependencies: coretext_dep)
+        deps += coretext_dep
+        conf.set('CONFIG_CORETEXT', 1)
+        coretext = true
+    elif cc.has_type('CTRunRef', prefix: '#include <ApplicationServices/ApplicationServices.h>', dependencies: appservices_dep)
+        deps += appservices_dep
+        conf.set('CONFIG_CORETEXT', 1)
+        coretext = true
+    elif get_option('coretext').enabled()
+        error('CoreText was enabled explicitly, but required headers or frameworks are missing.')
+    endif
+endif
+
+if get_option('require-system-font-provider') and not fontconfig and not directwrite and not coretext
+    error('''Either DirectWrite (on Windows), CoreText (on OSX), or Fontconfig
+(Linux, other) is required. If you really want to compile without
+a system font provider, set -Drequire-system-font-provider=false''')
+endif
+
+# ASM
+
+asm_args = []
+
+# Architecture check
+if host_machine.cpu_family() == 'x86'
+    bittype = '32'
+    asm_args += '-DARCH_X86_64=0'
+    cpu_family = 'x86'
+elif host_machine.cpu_family() == 'x86_64'
+    bittype = '64'
+    asm_args += '-DARCH_X86_64=1'
+    cpu_family = 'x86'
+elif host_machine.cpu_family() == 'aarch64' and host_system == 'darwin'
+    bittype = '64'
+    cpu_family = 'arm64'
+endif
+
+if cpu_family == 'x86'
+    nasm = find_program('nasm', required: get_option('asm'))
```

Comment:
Meson has native support for nasm. You can `add_language('nasm', required: get_option('asm'))` and `.asm` files will automatically be compiled with it.

Reply:
I'm trying to avoid bumping the version for now.

Reply:
nasm has been added in 0.64, that's really old. I don't see a reason for not depending on fairly new meson version since you'll still have autotools for legacy distro. Debian stable has meson 1.0.

Reply:
"Really old" is quite subjective. It's been out for "only" a year.

It's not available in oldstable, nor in Ubuntu LTS, nor in RHEL9. It's also newer than the most recent meson release to work on truly decrepit systems where python 3.6 is the only python (keeping in mind that even 3.7 is EOL, now).

How much effort do people wish to go, in order to support legacy systems? Good question (and one I usually try to avoid making value judgments about).

Reply:
For libass, broad compatibility is a very clear desire.

Separate from this particular project, I usually try to align with Ubuntu LTS when advocating for bumping minimums in OSS projects, as that's fairly defensible and means I usually won't have a long fight over the bump. 0.64, as Eli mentioned, does not meet that bar.

Reply:
Also probably relevant: the current muon release `0.2.0` claims compatibility with meson `0.64.x` which would already be enough for all the “future cleanup” tasks here. (And muon git master claims compatibility with meson `1.1.x`)